### ISSUE TITLE
Wrong mysql version installed on Ubuntu 16.04 (5.7 installed instead of 5.6)

### ISSUE THREAD

**Issue by johanvanhelden** [puphpet/puphpet#2495]
In my config I defined version 5.6, but it installed 5.7 (5.7.16-0ubuntu0.16.04.1 to be exact)

This is what the part config section looks like:

```
mysql:
    install: '1'
    settings:
        version: '5.6'
```

This happend with an archive I generated on puphpet.com today.

I am using the `puphpet/ubuntu1604-x64` box.

**Comment by pascalwacker** [comment]
I had the same issue. I tried to install MySQL version 5.6, but it tried to install 5.7, well actually, not even the installation on 5.7 worked for me.

The installation of 5.7 failed, as it said, it didn't have the PUBKEY for it.
To fix this, I had to login with SSH, run this command: `sudo apt-key adv --keyserver keyserver.ubuntu.com --recv-keys 8C718D3B5072E1F5`, `sudo apt-get update --fix-missing` and `sudo apt-get upgrade -y --force-yes` and then run a `vagrant provision` again. I also changed the dns settings to the Google DNS servers `8.8.8.8` and `8.8.4.4` not sure, if that helped too.

However, I now have MySQL 5.7 installed, but at least I have any MySQL installed.

Host: Ubuntu 16.10
Hypervisor: VMware Workstation

```
vagrantfile:
    target: local
    vm:
        provider:
            local:
                box: puphpet/ubuntu1604-x64
                box_url: puphpet/ubuntu1604-x64
                box_version: '0'
                chosen_virtualizer: vmware_workstation
                virtualizers:
                    virtualbox:
                        modifyvm:
                            natdnshostresolver1: false
                        showgui: 0
                    vmware:
                        numvcpus: 2
                    parallels:
                        linked_clone: 0
                        check_guest_tools: 0
                        update_guest_tools: 0
                machines:
                    vflm_nnb7fxvjlbk9:
                        id: ef
                        hostname: ef.local
                        network:
                            private_network: 192.168.57.101
                            forwarded_port:
                                vflmnfp_qkgnga1b61d2:
                                    host: '80'
                                    guest: '80'
                        memory: '2048'
                        cpus: '3'
        provision:
            puppet:
                manifests_path: puphpet/puppet/manifests
                module_path:
                    - puphpet/puppet/modules
                    - puphpet/puppet/manifests
                options:
                    - '--verbose'
                    - '--hiera_config /vagrant/puphpet/puppet/hiera.yaml'
        synced_folder: {  }
        usable_port_range:
            start: 10200
            stop: 10500
        post_up_message: ''
    ssh:
        host: 'false'
        port: 'false'
        private_key_path: 'false'
        username: vagrant
        guest_port: 'false'
        keep_alive: '1'
        forward_agent: 'false'
        forward_x11: 'false'
        shell: 'bash -l'
        insert_key: 'false'
    vagrant:
        host: detect
    proxy:
        http: ''
        https: ''
        ftp: ''
        no_proxy: ''
server:
    install: '1'
    packages:
        - curl
        - git
        - htop
        - acl
        - make
        - pkg-config
        - libmagickwand-dev
        - imagemagick
        - build-essential
        - libapache2-mod-php7.0
        - php-curl
users_groups:
    install: '1'
    groups: {  }
    users: {  }
locale:
    install: '1'
    settings:
        default_locale: de_CH.UTF-8
        locales:
            - de_CH.UTF-8
            - de_DE.UTF-8
            - en_GB.UTF-8
            - en_US.UTF-8
            - fr_CH.UTF-8
            - fr_FR.UTF-8
            - it_CH.UTF-8
            - it_IT.UTF-8
        timezone: Europe/Zurich
firewall:
    install: '1'
    rules: {  }
cron:
    install: '1'
    jobs: {  }
nginx:
    install: '0'
    settings:
        version: present
        default_vhost: 1
        proxy_buffers: '4 256k'
        proxy_buffer_size: 128k
        proxy_connect_timeout: 600s
        proxy_send_timeout: 600s
        proxy_read_timeout: 600s
        names_hash_bucket_size: 128
    upstreams: {  }
    vhosts:
        nxv_l1a8715fjrec:
            server_name: awesome.dev
            server_aliases:
                - www.awesome.dev
            www_root: /var/www/awesome
            listen_port: '80'
            client_max_body_size: 1m
            ssl: '0'
            ssl_cert: ''
            ssl_key: ''
            ssl_port: '443'
            ssl_protocols: ''
            ssl_ciphers: ''
            rewrite_to_https: '1'
            spdy: '1'
            locations:
                nxvl_82rb2jdix6si:
                    www_root: ''
                    location: /
                    index_files:
                        - index.html
                        - index.htm
                        - index.php
                    try_files:
                        - $uri
                        - $uri/
                        - /index.php$is_args$args
                    fastcgi: ''
                    fastcgi_index: ''
                    fastcgi_split_path: ''
                    proxy: ''
                    proxy_redirect: ''
                nxvl_mbqbo4knxy6d:
                    www_root: ''
                    location: '~ \.php$'
                    try_files:
                        - $uri
                        - $uri/
                        - /index.php$is_args$args
                    fastcgi: '127.0.0.1:9000'
                    fastcgi_index: index.php
                    fastcgi_split_path: '^(.+\.php)(/.*)$'
                    fast_cgi_params_extra:
                        - 'SCRIPT_FILENAME $request_filename'
                        - 'APP_ENV dev'
                    set:
                        - '$path_info $fastcgi_path_info'
                    proxy: ''
                    proxy_redirect: ''
    proxies: {  }
apache:
    install: '1'
    settings:
        version: 2.4
        user: www-data
        group: www-data
        default_vhost: true
        manage_user: false
        manage_group: false
        sendfile: 0
    modules:
        - cache
        - deflate
        - expires
        - headers
        - info
        - mime_magic
        - proxy_fcgi
        - rewrite
        - userdir
        - vhost_alias
    vhosts:
        av_jsovp7rlezdw:
            servername: ef.local
            serveraliases:
                - www.ef.local
            docroot: /var/www/html
            port: '80'
            setenv:
                - 'APP_ENV dev'
            setenvif:
                - 'Authorization "(.*)" HTTP_AUTHORIZATION=$1'
            custom_fragment: ''
            ssl: '0'
            ssl_cert: ''
            ssl_key: ''
            ssl_chain: ''
            ssl_certs_dir: ''
            ssl_protocol: ''
            ssl_cipher: ''
            directories:
                avd_ay86bgppv1n0:
                    path: /var/www/html
                    options:
                        - Indexes
                        - FollowSymlinks
                        - MultiViews
                    allow_override:
                        - All
                    require:
                        - 'all granted'
                    custom_fragment: ''
                    files_match:
                        avdfm_o42e681tj4vu:
                            path: \.php$
                            sethandler: 'proxy:fcgi://127.0.0.1:9000'
                            custom_fragment: ''
                            provider: filesmatch
                    provider: directory
letsencrypt:
    install: '1'
    settings:
        email: ''
    domains: {  }
php:
    install: '1'
    settings:
        version: '7.0'
    modules:
        php:
            - cli
            - intl
            - xml
            - mcrypt
            - curl
            - gd
            - mysql
            - mysqli
            - xsl
        pear: {  }
        pecl: {  }
    ini:
        display_errors: 'On'
        session.save_path: /var/lib/php/session
        date.timezone: Europe/Zurich
        session.gc_maxlifetime: '1800'
        memory_limit: 256M
        max_execution_time: '60'
        max_input_time: '90'
        upload_max_filesize: 20M
        post_max_size: 24M
        error_reporting: 'E_ALL & ~E_STRICT'
    fpm_ini:
        error_log: /var/log/php-fpm.log
    fpm_pools:
        phpfp_cqz3wv8e2skh:
            ini:
                prefix: www
                listen: '127.0.0.1:9000'
                security.limit_extensions: .php
                user: vagrant
                group: vagrant
    composer: '1'
    composer_home: ''
xdebug:
    install: '1'
    settings:
        xdebug.default_enable: '1'
        xdebug.remote_autostart: '0'
        xdebug.remote_connect_back: '1'
        xdebug.remote_enable: '1'
        xdebug.remote_handler: dbgp
        xdebug.remote_port: '9090'
blackfire:
    install: '0'
    settings:
        server_id: ''
        server_token: ''
        agent:
            http_proxy: ''
            https_proxy: ''
            log_file: stderr
            log_level: '1'
        php:
            agent_timeout: '0.25'
            log_file: ''
            log_level: '1'
xhprof:
    install: '0'
wpcli:
    install: '0'
    version: v0.24.1
drush:
    install: '0'
    version: 8.0.5
ruby:
    install: '1'
    versions:
        rv_960tdbbkt8k5:
            default: '1'
            bundler: '1'
            version: 2.3.1
            gems:
                - deep_merge@1.0.1
                - activesupport@4.2.6
                - vine@0.2
python:
    install: '1'
    packages: {  }
    versions: {  }
nodejs:
    install: '0'
    settings:
        version: '6'
    npm_packages: {  }
hhvm:
    install: '0'
    composer: '1'
    composer_home: ''
    settings: {  }
    server_ini:
        hhvm.server.host: 127.0.0.1
        hhvm.server.port: '9000'
        hhvm.log.use_log_file: '1'
        hhvm.log.file: /var/log/hhvm/error.log
    php_ini:
        display_errors: 'On'
        error_reporting: '-1'
        date.timezone: UTC
mariadb:
    install: '0'
    settings:
        version: '10.1'
        root_password: '123'
        override_options: {  }
    adminer: 0
    users:
        mariadbnu_ksaxe6eg0bke:
            name: dbuser
            password: '123'
    databases:
        mariadbnd_2kinr3u6ysmi:
            name: dbname
            sql: ''
    grants:
        mariadbng_aw91f9vuxsn5:
            user: dbuser
            table: '*.*'
            privileges:
                - ALL
mysql:
    install: '1'
    settings:
        version: '5.6'
        root_password: '123'
        override_options: {  }
    adminer: 0
    users:
        mysqlnu_nrrs995jzman:
            name: dev
            password: '123'
    databases:
        mysqlnd_jwti4fshariq:
            name: dev
            sql: ''
    grants:
        mysqlng_3oyba0tb9jdl:
            user: dev
            table: '*.*'
            privileges:
                - ALL
postgresql:
    install: '0'
    settings:
        global:
            encoding: UTF8
            version: '9.6'
        server:
            postgres_password: '123'
    databases: {  }
    users: {  }
    grants: {  }
    adminer: 0
mongodb:
    install: '0'
    settings:
        bind_ip: 127.0.0.1
        port: '27017'
    globals:
        version: 2.6.0
    databases: {  }
redis:
    install: '0'
    settings:
        port: '6379'
sqlite:
    install: '0'
    adminer: 0
    databases: {  }
mailhog:
    install: '0'
    settings:
        smtp_ip: 0.0.0.0
        smtp_port: 1025
        http_ip: 0.0.0.0
        http_port: '8025'
        path: /usr/local/bin/mailhog
beanstalkd:
    install: '0'
    settings:
        listenaddress: 0.0.0.0
        listenport: '11300'
        maxjobsize: '65535'
        maxconnections: '1024'
        binlogdir: /var/lib/beanstalkd/binlog
        binlogfsync: null
        binlogsize: '10485760'
    beanstalk_console: 0
rabbitmq:
    install: '0'
    settings:
        port: '5672'
    users: {  }
    vhosts: {  }
    plugins: {  }
elastic_search:
    install: '0'
    settings:
        version: 2.3.1
        java_install: true
    instances:
        esi_3f3edov87cnd:
            name: es-01
solr:
    install: '0'
    settings:
        version: 5.5.2
        port: '8984'
```

**Comment by holodoc** [comment]
Same thing happening to me on 16.04 LTS. Even though 5.6 is clearly used in `config.yaml` the provisioned machine has MySQL 5.7 installed.

**Comment by monnsena** [comment]
hmm.. looks like same issue with me.

**Comment by n9iels** [comment]
Same issue over here. I created a config with help of puphpet.com and MySQL 5.6 is specified in the `config.yaml`, but MySQL 5.7 is installed.

However I think that puphpet is not the main problem. It looks like that MySQL 5.6 is not present on Ubuntu 16 machines. You have to include the Ubuntu 14 repository to install it. See this topic for example: http://askubuntu.com/questions/762384/install-mysql-5-6-on-ubuntu-16-04

**Comment by intnick** [comment]
I tried adding a file to files/exec-preprovision/ with:
#!/bin/bash
sudo add-apt-repository 'deb http://archive.ubuntu.com/ubuntu trusty universe'
sudo apt-get update

I can do a manual mysql 5.6 instalation when I log in to the system now, but I'm still getting 5.7 installed initially, can't figure out why, because the execution of the script happens much before mysql is added to the system by puppet.

**Comment by jtreminio** [comment]
Fixed in 73dfa5354375f5265b4231561fa3b8f7c6453ab9
